Add interface to MigrationInterceptorProvider

darrenhull opened the pull request:
I have added this interface to allow the functionality to be override and extended. for my project we have inheritance within the documents meaning that the runtime migrations can really mess things up. So this allows me to but a noop serializer in.

darrenhull committed:
Add interface to MigrationInterceptorProvider

SRoddis commented:
Hey darrenhull,

nice work. Your Pull Request results in a failing build. Did you run the tests? Please run the test and fix the Problem. Looks like the interface cant be resolved in the test. When you fix the build i cant merge the pull request.

https://ci.appveyor.com/project/SRoddis/mongo-migration/builds/32226216

Cheers, 
Sean

SRoddis commented on Mongo.Migration/Migrations/Locators/MigrationLocator.cs:
```diff
@@ -77,7 +77,8 @@ public DocumentVersion GetLatestVersion(Type type)
 
-            return migrations.Max(m => m.Version);
+            return migrations?.Max(m => m.Version)
+                ?? new DocumentVersion(0,0,0);
```
Use DocumentVersion default for this. Its a struct. 

` return migrations?.Max(m => m.Version) ?? DocumentVersion.Default();`

SRoddis replied:
I needet to add this line of code to fix another issue. You can remove it from the PR.

darrenhull committed:
Fix test issue when resolving IMigrationInterceptorProvider

darrenhull commented:
I have removed the migrations?.Max(m => m.Version)
                ?? new DocumentVersion(0,0,0); and fixed the tests. (I think couldnt get netstandard 2.0 to run but 2.2 ran and passed. will check appveyer result.)

darrenhull commented:
The tests passed but looks like it is missing an Environment variable which resulted in a failure:
No token found in Environment Variable 'COVERALLS_REPO_TOKEN'.
Command exited with code 1

SRoddis commented:
Yes, i changed the token. That has nothing todo with your commit. I merged your pull request, please update to 3.0.112

Cheers, Sean
